examples/fabric-sync/bridge/src/FabricBridge.cpp

```diff
@@ -0,0 +1,219 @@
+{
+    return sInstance;
+}
+
+CHIP_ERROR FabricBridge::AddSynchronizedDevice(const SynchronizedDevice & data)
+{
+    ChipLogProgress(NotSpecified, "Received AddSynchronizedDevice: Id=[%d:" ChipLogFormatX64 "]", data.id.GetFabricIndex(),
+                    ChipLogValueX64(data.id.GetNodeId()));
+
+    // Create a new BridgedDevice and set it as reachable
+    auto device = std::make_unique<BridgedDevice>(data.id);
+    device->SetReachable(true);
+
+    // Initialize BridgedDevice attributes from data
+    BridgedDevice::BridgedAttributes attributes;
+
+    if (data.uniqueId.has_value())
+    {
+        attributes.uniqueId = data.uniqueId.value();
+    }
+
+    if (data.vendorName.has_value())
+    {
+        attributes.vendorName = data.vendorName.value();
+    }
+
+    if (data.vendorId.has_value())
+    {
+        attributes.vendorId = data.vendorId.value();
+    }
+
+    if (data.productName.has_value())
+    {
+        attributes.productName = data.productName.value();
+    }
+
+    if (data.productId.has_value())
+    {
+        attributes.productId = data.productId.value();
+    }
+
+    if (data.nodeLabel.has_value())
+    {
+        attributes.nodeLabel = data.nodeLabel.value();
+    }
+
+    if (data.hardwareVersion.has_value())
+    {
+        attributes.hardwareVersion = data.hardwareVersion.value();
+    }
+
+    if (data.hardwareVersionString.has_value())
+    {
+        attributes.hardwareVersionString = data.hardwareVersionString.value();
+    }
+
+    if (data.softwareVersion.has_value())
+    {
+        attributes.softwareVersion = data.softwareVersion.value();
+    }
+
+    if (data.softwareVersionString.has_value())
+    {
+        attributes.softwareVersionString = data.softwareVersionString.value();
+    }
+
+    // Set bridged device attributes and ICD status
+    device->SetBridgedAttributes(attributes);
+    device->SetIcd(data.isIcd.value_or(false));
+
+    // Add the device to the bridge manager with a parent endpoint
+    auto result = BridgeDeviceMgr().AddDeviceEndpoint(std::move(device), /* parentEndpointId= */ 1);
+    if (!result.has_value())
+    {
+        ChipLogError(NotSpecified, "Failed to add device with Id=[%d:0x" ChipLogFormatX64 "]", data.id.GetFabricIndex(),
+                     ChipLogValueX64(data.id.GetNodeId()));
+        return CHIP_ERROR_ENDPOINT_POOL_FULL;
+    }
+
+    // Retrieve and verify the added device by ScopedNodeId
+    BridgedDevice * addedDevice = BridgeDeviceMgr().GetDeviceByScopedNodeId(data.id);
+    VerifyOrDie(addedDevice);
+
+    ChipLogProgress(NotSpecified, "Added device with Id=[%d:0x" ChipLogFormatX64 "]", data.id.GetFabricIndex(),
+                    ChipLogValueX64(data.id.GetNodeId()));
+
+    // Add the Ecosystem Information Cluster to the device's endpoint
+    CHIP_ERROR err =
+        app::Clusters::EcosystemInformation::EcosystemInformationServer::Instance().AddEcosystemInformationClusterToEndpoint(
+            addedDevice->GetEndpointId());
```

Comment:
nit, would it be log if there is an error returned from AddEcosystemInformationClusterToEndpoint